File: src/main/java/frc/robot/commands/Intake_Commands/AutoArm.java
package frc.robot.commands.Intake_Commands;

import com.revrobotics.CANSparkBase;
import edu.wpi.first.wpilibj2.command.Command;
import frc.robot.Constants;
import frc.robot.subsystems.Intake;

public class AutoArm extends Command {
    private double targetTraj = 0.0;
    private final Intake intake;
    private boolean isDown = false;

    public AutoArm(Intake intake, boolean isDown) {
        this.intake = intake;
        this.isDown = isDown;
        if (isDown) {
            targetTraj = Constants.INTAKE_ARM_DOWN;
        } else {
            targetTraj = Constants.INTAKE_ARM_HOLD;
        }
        addRequirements(intake);
    }

    @Override
    public void initialize() {
        System.out.println("INFO: AutoArm.initialize");
        intake.getRaiseLowerSpinner().getPIDController().setReference(targetTraj, CANSparkBase.ControlType.kSmartMotion);
    }

    @Override
    public void execute() {
//        intake.raiseLowerSpinner.pidController.setReference(targetTraj, CANSparkMax.ControlType.kSmartMotion)
    }

    @Override
    public boolean isFinished() {
        var trajPoseError = targetTraj - intake.getRaiseLowerSpinner().getEncoder().getPosition();
        if (isDown) {
            return Math.abs(trajPoseError) < 1;
        } else {
            return false;
        }
    }

    @Override
    public void end(boolean interrupted) {
        System.out.println("INFO: AutoArm.end");
        intake.stopArm();
    }
}

File: src/main/java/frc/robot/commands/Intake_Commands/AutoArm.kt
package frc.robot.commands.Intake_Commands

import com.revrobotics.CANSparkMax
import edu.wpi.first.wpilibj2.command.CommandBase
import frc.robot.Constants
import frc.robot.subsystems.Intake
import kotlin.math.abs

class AutoArm(private val intake: Intake, private val isDown: Boolean) : CommandBase() {
    private var targetTraj = 0.0

    init {
        addRequirements(intake)

        targetTraj = if (isDown) {
                Constants.INTAKE_ARM_DOWN
            } else {
               Constants.INTAKE_ARM_HOLD
            }

    }

    override fun initialize() {
        println("INFO: TargetTraj: $targetTraj")
        intake.raiseLowerSpinner.pidController.setReference(targetTraj, CANSparkMax.ControlType.kSmartMotion)
    }

    override fun execute() {
//        intake.raiseLowerSpinner.pidController.setReference(targetTraj, CANSparkMax.ControlType.kSmartMotion)
    }

    override fun isFinished(): Boolean {
        val trajPoseError = targetTraj - intake.raiseLowerSpinner.encoder.position
        return if (isDown) {
            abs(trajPoseError) < 1
        } else {
            false
        }
    }

    override fun end(interrupted: Boolean) {
        println("INFO: Intake Arm Down")
        intake.stopArm()
    }
}